expyfun/stimuli/_tracker.py

```diff
@@ -283,17 +283,15 @@ def respond(self, correct):
         else:
             self._reversals = np.append(self._reversals, 0)
 
-        if bad:
-            self._bad_reversals = np.append(self._bad_reversals,
-                                            self._n_reversals)
+        self._bad_reversals = np.append(self._bad_reversals, bad)
 
         # Should we stop here?
         self._stopped = self._stop_here()
 
         if not self._stopped:
-            if self._x_min is not None:
+            if self._x_min is not -np.inf:
                 self._x[-1] = max(self._x_min, self._x[-1])
-            if self._x_max is not None:
+            if self._x_max is not np.inf:
                 self._x[-1] = min(self._x_max, self._x[-1])
```

Comment:
Ahh I see, it's done here. Why not move these lines up to where the `self._x_max` / `self._x_min` checks have already been done? Seems a bit cleaner to do it all at once.